doc/chapters/api/index.rst

```diff
@@ -32,102 +32,187 @@ BindMount
 
-:Parameters:
-        :config: ``string`` config file in json format.
+Parameters
+##########
+* config: ``string`` config file in json format.
+
+Example config JSON::
 
-                Example config JSON::
+[{"enableWriteBuffer": true}]
 
-                [{"enableWriteBuffer": true}]
+Return Values
+#############
+* containerID: ``int32`` ID of created SoftwareContainer.
 
-|
+Error sources
+#############
+* Network setup issues
+    * Bridge device not available and unable to set it up
+    * Given netmask can not be used to assign IP address to container
 
-:Return Values:
-        :containerID: ``int32`` ID of created SoftwareContainer.
+* LXC issues
+    * Couldn't prepare file system while initializing container
+    * Container ID already being in use (caused by lingering old containers)
+    * LXC config couldn't be loaded
+    * LXC could not create container
 
 Destroy
 -------
 Tears down all active gateways related to container and shuts down the
 container with all reserved sources.
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+
+Error sources
+#############
+* Invalid ID: No matching container exists.
+* Invalid state: The container is not ready or is suspended
+* Gateway errors: One or more of the gateways might have failed to shut down
+* LXC issues:
+    * LXC could not shutdown the container
+    * LXC could not force stop the container (if shutdown fails)
+    * LXC could not destroy the container
 
-|
+Failing to destroy the container leads to it being put into an invalid state. One should
+(ironically) destroy such a container.
 
 .. _dbus-execute:
 
 Execute
 -------
 Launches the specified application/code in the container.
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
-        :commandLine: ``string`` the method to run in container.
-        :workDirectory: ``string`` path to working directory.
-        :outputFile: ``string`` output file to direct stdout.
-        :env: ``map<string, string>`` environment variables and their values.
-
-|
-
-:Return Values:
-        :pid: ``int32`` PID of process running in the container, as seen by the host.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+* commandLine: ``string`` the method to run in container.
+* workDirectory: ``string`` path to working directory.
+* outputFile: ``string`` output file to direct stdout.
+* env: ``map<string, string>`` environment variables and their values.
+
+Return value
+############
+* pid: ``int32`` PID of process running in the container, as seen by the host.
+
+Error sources
+#############
+* Bad state: The container is not ready or is suspended
+* Capability error: couldn't set default capabilities (if ``SetCapabilities`` has not already been
+  called)
+* LXC error: The call to ``attach()`` failed.
```

Comment:
I think this is too detailed information about the interaction with LXC. It's a maintenance burden potentially (an unnecessary one). What about "the underlying LXC operation failed" or something like that?